pandaserver/dataservice/ddm.py

```diff
@@ -0,0 +1,865 @@
+            if lifetime is not None:
+                client.set_metadata(scope, dataset_name, key="lifetime", value=lifetime * 86400)
+        except DataIdentifierAlreadyExists:
+            pass
+        # open dataset just in case
+        try:
+            client.set_status(scope, dataset_name, open=True)
+        except Exception:
+            pass
+        # add files
+        if len(files) > 0:
+            i_files = 0
+            n_files = 1000
+            while i_files < len(files):
+                tmp_files = files[i_files : i_files + n_files]
+                try:
+                    client.add_files_to_dataset(scope=scope, name=dataset_name, files=tmp_files, rse=None)
+                except FileAlreadyExists:
+                    for tmp_file in tmp_files:
+                        try:
+                            client.add_files_to_dataset(scope=scope, name=dataset_name, files=[tmp_file], rse=None)
+                        except FileAlreadyExists:
+                            pass
+                i_files += n_files
+        # Format is a 36-character string divided into five groups separated by hyphens. The groups have 8, 4, 4, 4, and 12 characters.
+        # After the formatting, the vuid string would look something like this: 12345678-1234-1234-1234-123456789012
+        vuid = hashlib.md5((scope + ":" + dataset_name).encode()).hexdigest()
+        vuid = f"{vuid[0:8]}-{vuid[8:12]}-{vuid[12:16]}-{vuid[16:20]}-{vuid[20:32]}"
+        duid = vuid
+        return {"duid": duid, "version": 1, "vuid": vuid}
+
+    def register_dataset_location(
+        self,
+        dataset_name: str,
+        rses: list,
+        lifetime: int = None,
+        owner: str = None,
+        activity: str = None,
+        scope: str = None,
+        grouping: str = "DATASET",
+        notify: str = "N",
+    ) -> bool:
+        """
+        Register a dataset location in Rucio
+
+        Parameters:
+        dataset_name (str): Dataset name
+        rses (list): List of RSEs
+        lifetime (int, optional): Lifetime of the dataset in seconds. Defaults to None.
+        owner (str, optional): Owner of the dataset. Defaults to None.
+        activity (str, optional): Activity associated with the dataset. Defaults to None.
+        scope (str, optional): Scope of the dataset. Defaults to None.
+        grouping (str, optional): Grouping of the dataset. Defaults to "DATASET".
+        notify (str, optional): Notification option. Defaults to "N".
+
+        Returns:
+        bool: True if the operation is successful, False otherwise
+        """
+        grouping = "DATASET" if grouping is None else grouping
+        preset_scope = scope
+        lifetime = lifetime * 24 * 60 * 60 if lifetime else None
+        scope, dataset_name = self.extract_scope(dataset_name)
+        scope = preset_scope if preset_scope else scope
+        dids = []
+        did = {"scope": scope, "name": dataset_name}
+        dids.append(did)
```

Comment:
This could be directly
did = {"scope": scope, "name": dataset_name}
dids = [did]